### DISCUSSION TITLE
Questions about combining model options

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#2438] [Q&A]
Hi. I'm simulating an experiment that's long enough to have a cycle of over 100. (https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/simulating-long-experiments.ipynb)

I want to combine thermal option({"thermal : x-lumped"}) and aging option( {"SEI": "ec reaction limited"}) together.
However, looking at the plot, the battery temperature does not change.
![Figure_1;;](https://user-images.githubusercontent.com/113442395/200586516-cadfb496-83c8-4eb4-8de2-28d61df898e9.png)
I want to change the temperature of the battery, but what should I do?
I'd really appreciate your help.

Here's my code.

import pybamm
import matplotlib.pyplot as plt
import numpy as np

parameter_values = pybamm.ParameterValues("Mohtat2020")
parameter_values.update({"SEI kinetic rate constant [m.s-1]": 1e-14})

**spm = pybamm.lithium_ion.SPM({"SEI": "ec reaction limited"}, {"thermal : x-lumped"})**
# Calculate stoichiometries at 100% SOC
print(parameter_values)
param = spm.param
esoh_solver = pybamm.lithium_ion.ElectrodeSOHSolver(parameter_values, param)

Vmin = 3.0
Vmax = 4.2
Cn = parameter_values.evaluate(param.n.cap_init)
Cp = parameter_values.evaluate(param.p.cap_init)
n_Li_init = parameter_values.evaluate(param.n_Li_particles_init)

inputs={ "V_min": Vmin, "V_max": Vmax, "C_n": Cn, "C_p": Cp, "n_Li": n_Li_init}
esoh_sol = esoh_solver.solve(inputs)

print(f"Initial negative electrode SOC: {esoh_sol['x_100'].data[0]:.3f}")
print(f"Initial positive electrode SOC: {esoh_sol['y_100'].data[0]:.3f}")

# Update parameter values with initial conditions
c_n_max = parameter_values.evaluate(param.n.prim.c_max)
c_p_max = parameter_values.evaluate(param.p.prim.c_max)
parameter_values.update(
    {
        "Initial concentration in negative electrode [mol.m-3]": esoh_sol["x_100"].data[0] * c_n_max,
        "Initial concentration in positive electrode [mol.m-3]": esoh_sol["y_100"].data[0] * c_p_max,
    }
)
pybamm.set_logging_level("NOTICE")

experiment = pybamm.Experiment([
    (f"Discharge at 4C until 3.0 V",
     #"Hold at 3.0 V until C/50",
     "Rest for 1 hour",
    f"Charge at 3 C until 4.1 V",
    f"Hold at 4.1 V until C/50")
]*800 ,
termination="20% capacity"
)
sim = pybamm.Simulation(spm, experiment=experiment, parameter_values=parameter_values)
sol = sim.solve()
#sol.plot(["Current [A]", "Terminal voltage [V]"])
pybamm.plot_summary_variables(sol)
sol_list = sim.solve(save_at_cycles=[30,45,55])
sol_list.cycles[44].plot(["Current [A]","Terminal voltage [V]","X-averaged cell temperature [K]","Discharge capacity [A.h]"])

**Comment by valentinsulzer** [comment]
pybamm.lithium_ion.SPM({"SEI": "ec reaction limited", "thermal : x-lumped"})

**Reply by YouHyin** [reply]
Yes, I also entered the code like this. (If you look at my code, you can see that I typed it like this(spm = pybamm.lithium_ion.SPM({"SEI": "ec reaction limited"}, {"thermal : x-lumped"}).

**Reply by rtimms** [reply]
You have an extra set of braces between "ec reaction limited" and "thermal"

**Reply by YouHyin** [reply]
Okay, Thank you!
I tried to execute the code after modifying it as you told me. However, even if you keep running the code over and over again, it keeps stopping and not working. How do I solve this?
![1234561531](https://user-images.githubusercontent.com/113442395/200844480-5b1b0090-7fa5-4b11-b531-666d6348fc76.png)
And thank you so much for letting me know!

**Reply by YouHyin** [reply]
There is also one more question.
When **"Experiment is infeasible: 'event: Maximum voltage"** pops up, I want to ignore it and continue to run the simulation without stopping. What should I do?
![1234568](https://user-images.githubusercontent.com/113442395/200864061-fb0e571d-7e14-482d-b42b-2816c76fa49b.png)

**Reply by valentinsulzer** [reply]
> "Experiment is infeasible: 'event: Maximum voltage"

Which version are you using? There has been an update in v22.10 to (hopefully) stop this from happening

**Reply by Thiyaga2025** [reply]
@valentinsulzer and @YouHyin,
i have a question regarding below part of the above mentioned code,

![image](https://github.com/user-attachments/assets/94909768-d765-461c-b31b-a21de58f7423)

won't this be done by PyBamm itself, should we manually feed this information. when i tested with and without this particular step there is some minor difference in initial concentration. please help me here. thanks.

**Reply by valentinsulzer** [reply]
This code is updating the initial SOC by changing the initial concentrations. You can also make pybamm do this by passing "initial_soc=x" to sim.solve(). (x should be a float between 0 and 1).

**Reply by Thiyaga2025** [reply]
hi @valentinsulzer , thanks a lot for the clarification. i guess, in that case it is equivalent to setting initial_soc=1. in sim.solve() and the user can do it in either way(in initial_soc or through the above mentioned process). what's the significance of doing this using eSOH solver since it is very easy to set "initial_soc" value in sim.solve(). also, how to set an initial_soc value of 0.5(or any value other than 1 by eSOH solver approach. can i simply multiply the 100% concentration with the fraction value. regards.

**Reply by valentinsulzer** [reply]
There is no special significance, but sometimes people like to know what's happening inside a function.
You are correct about `initial_soc=1`. To set a different initial SOC, you need to do `x = x_0 + (x_100 - x_0) * soc`, and the same thing for `y` (note that `y_0 > y_100` but the same equation works)

**Reply by Thiyaga2025** [reply]
thanks @valentinsulzer , this helps.
